### PR TITLE
mam2: various cleanups

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: mam/v2/prng/prng.h [hunk 1]
```diff
@@ -44,7 +44,8 @@ retcode_t mam_prng_destroy(mam_prng_t *const prng);
  * @param nonce The nonce
  * @param output Pseudorandom output trits
  */
-void mam_prng_gen(mam_prng_t const *const prng, tryte_t const destination,
+void mam_prng_gen(mam_prng_t const *const prng,
+                  prng_destination_tryte_t const destination,
```
Comment (thibault-martinez) [inline mam/v2/prng/prng.h]: Can we rename this to `mam_prng_dest_tryte_t` ?

File: mam/v2/mss/tests/test_mss.c [hunk 2]
```diff
@@ -204,33 +204,35 @@ static bool mss_test(mss_t *mss, mam_prng_t *prng, mam_spongos_t *spongos,
 
     do {
       mss_sign(mss, hash, sig);
-      r = r && mss_verify(spongos, &wots->spongos, hash, sig, pk);
+      mam_spongos_t wots_spongos;
```
Comment (thibault-martinez) [inline mam/v2/mss/tests/test_mss.c]: Can you declare this before the loop ?

File: mam/v2/mam/endpoint.c [hunk 3]
```diff
@@ -165,21 +163,15 @@ size_t mam_endpoints_serialized_size(mam_endpoint_t_set_t const endpoints) {
...
 
   HASH_ITER(hh, endpoints, entry, tmp) {
-    if ((ret = mam_endpoint_serialize(&entry->value, buffer)) !=
-        RC_OK) {  // endpoint
-      return ret;
-    }
+    mam_endpoint_serialize(&entry->value, buffer);
   }
-
-  return ret;
```
Comment (thibault-martinez) [inline mam/v2/mam/endpoint.c:182]: This function still returns `retcode_t`

File: mam/v2/mam/channel.c [hunk 4]
```diff
@@ -180,23 +180,17 @@ size_t mam_channels_serialized_size(mam_channel_t_set_t const channels) {
...
 
   pb3_encode_size_t(mam_channel_t_set_size(channels),
                     buffer);  // channels number
 
   HASH_ITER(hh, channels, entry, tmp) {
-    if ((ret = mam_channel_serialize(&entry->value, buffer)) !=
-        RC_OK) {  // channel
-      return ret;
-    }
+    mam_channel_serialize(&entry->value, buffer);
```
Comment (thibault-martinez) [inline mam/v2/mam/channel.c:192]: Can you put back the `// channel` comment ?

File: mam/v2/mam/endpoint.c [hunk 5]
```diff
@@ -165,21 +163,15 @@ size_t mam_endpoints_serialized_size(mam_endpoint_t_set_t const endpoints) {
...
 
   pb3_encode_size_t(mam_endpoint_t_set_size(endpoints),
                     buffer);  // endpoints number
 
   HASH_ITER(hh, endpoints, entry, tmp) {
-    if ((ret = mam_endpoint_serialize(&entry->value, buffer)) !=
-        RC_OK) {  // endpoint
-      return ret;
-    }
+    mam_endpoint_serialize(&entry->value, buffer);
```
Comment (thibault-martinez) [inline mam/v2/mam/endpoint.c:173]: Can you put back the `// endpoint` comment ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: mam/v2/mam/endpoint.c (hunk 3 above)
Comment (tsvisabo) [reply mam/v2/mam/endpoint.c:182]: because it uses `mam_channel_t_set_add` which allocates and can fail

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**